### PR TITLE
Add reporting for BYTES_SENT by PXF

### PR DESCRIPTION
**Pull request by ashuka24**
This commit adds byte reporting at query and fragment level for Text,
BufferWritable, and GPDBWritable.

Co-authored-by: Ashuka Xue <axue@vmware.com>
Co-authored-by: Bradford Boyle <bradfordb@vmware.com>
Co-authored-by: Alex Denissov <adenissov@vmware.com>

### CODE REVIEW

File: server/pxf-api/src/main/java/org/greenplum/pxf/api/io/GPDBWritable.java [hunk 1]
```diff
@@ -465,7 +465,8 @@ public void write(DataOutput out) throws IOException {
                     /* For BYTEA format, add 4byte length header at the beginning  */
                     case BYTEA:
                         out.writeInt(colLength[i]);
-                        out.write((byte[]) colValue[i]);
+                        byte[] value = (byte[]) colValue[i];
+                        out.write(value);
```
Comment (denalex) [inline server/pxf-api/src/main/java/org/greenplum/pxf/api/io/GPDBWritable.java:469]: can collapse back to 1 line

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/MetricsReporter.java [hunk 2]
```diff
@@ -140,8 +140,9 @@ private Tags getTags(RequestContext context) {
      */
     public enum PxfMetric {
         FRAGMENTS_SENT("fragments.sent", "pxf.metrics.fragments.enabled", null),
-        RECORDS_SENT("records.sent", "pxf.metrics.records.enabled", "pxf.metrics.records.report-frequency"),
-        RECORDS_RECEIVED("records.received", "pxf.metrics.records.enabled", "pxf.metrics.records.report-frequency");
+        RECORDS_SENT("records.sent", "pxf.metrics.records.enabled", "pxf.metrics.report-frequency"),
+        RECORDS_RECEIVED("records.received", "pxf.metrics.records.enabled", "pxf.metrics.report-frequency"),
+        BYTES_SENT("bytes.sent", "pxf.metrics.bytes.enabled", "pxf.metrics.bytes.report-frequency");
```
Comment (denalex) [inline server/pxf-service/src/main/java/org/greenplum/pxf/service/MetricsReporter.java:145]:
```suggestion
        BYTES_SENT("bytes.sent", "pxf.metrics.bytes.enabled", "pxf.metrics.report-frequency");
```

**Review by denalex — COMMENTED** [review]
While the PR looks good, maybe this implementation should be changed in favor of using `https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/output/CountingOutputStream.html` like we are planning to do for the bytes received.

**Comment by ashuka24** [comment]
Closed in favor of: https://github.com/greenplum-db/pxf/pull/595

**Pull request closed without merging**